### PR TITLE
DONOT MERGE: Fix Sync Client and Examples

### PR DESCRIPTION
**Pull request by dhoomakethu**
1. Fix sync client and examples.
2. Fix repl server errors

### CODE REVIEW

**Comment by dhoomakethu** [comment]
TODO. FIx asyncio serial server.

File: pymodbus/repl/server/main.py [hunk 1]
```diff
@@ -45,7 +46,7 @@ class ModbusFramerTypes(str, Enum):
     binary = "binary"  # pylint: disable=invalid-name
 
 
-def _completer(incomplete: str, valid_values: list[str]) -> list[str]:
+def _completer(incomplete: str, valid_values: List[str]) -> List[str]:
```
Comment (janiversen) [inline pymodbus/repl/server/main.py:49]: please have a look at your pylint errors:
``
pymodbus/repl/server/main.py:64:32: R6002: 'typing.List' will be deprecated with PY39, consider using 'list' instead (consider-using-alias)
``
I am also running with 3.10 so using typing.List breaks all.

Add
```
from __future__ import annotations
```
at the top of the file and all problems are gone.

**Review by janiversen — CHANGES_REQUESTED** [review]
Please do not use typing.List see in pylint why.

File: pymodbus/repl/server/main.py (hunk 1 above)
Comment (dhoomakethu) [reply pymodbus/repl/server/main.py:49]: Yes, using list directly breaks things in 3.8. I will push those changes later tomorrow

Comment (janiversen) [reply pymodbus/repl/server/main.py:49]: Something else is breaking your code (unless you do not have the future import), I have been using "list" for years now in 3.8 - 3.9 and more lately 3.10 and 3.11

My production is running 3.8 for both Pymodbus and Homeassistant (where typing.List have been banned for more than y year).

Looking forward to see your push.

Comment (dhoomakethu) [reply pymodbus/repl/server/main.py:49]: Actually the import is already in there , refer https://github.com/riptideio/pymodbus/pull/1019/files#diff-68584e8d04ed7d3835b548b4f06e533dddb6e3e5ea49c4fff2ccfb0006791cfaR2

**Comment by janiversen** [comment]
A fast test shows your code passes pylint with python 3.10

**Review by janiversen — APPROVED** [review]

**Pull request merged**